### DISCUSSION TITLE
reducer protocol

### DISCUSSION THREAD

**Discussion by xuanyuelin** [pointfreeco/swift-composable-architecture#2711] [Beta]
when will the protocol-beta merged into the master?
I see the protocol-beta has not been updated for one year, but my old swift-ui project depends on the reducer protocols.
If the protocol-beta branch will not maintain any more, should I change to other branchs and change the old realities?

**Comment by kamcma** [comment]
#1283 was merged and released in 0.41.0. Then, in 1.0, the old `Reducer` type was removed and `ReducerProtocol` was renamed `Reducer`.

The official recommendation is that you upgrade to 0.58.0..<1.0.0 _first_. It will give you deprecation warnings and compiler-fix-its for renames and such. Then, once you have fixed all warnings and errors, you can upgrade to 1.0.0..<2.0.0.

**Reply by xuanyuelin** [reply]
Tks. I got it.
